build.gradle

```diff
@@ -540,5 +578,354 @@ wrapper {
+                                    }
+                                }
+                            }
+                            configuration {
+                                skipProvidedScope 'true'
+                                format 'html'
+                            }
+                        }
+                        plugin {
+                            groupId 'org.apache.maven.plugins'
+                            artifactId 'maven-compiler-plugin'
+                            version '3.8.0'
+                            configuration {
+                                source '11'
+                                target '11'
+                                debug 'true'
+                                debuglevel 'lines,vars,source'
+                                compilerArgument '-proc:none'
+                            }
+                        }
+                        plugin {
+                            groupId 'org.apache.maven.plugins'
+                            artifactId 'maven-surefire-plugin'
+                            version '2.22.1'
+                            configuration {
+                                skipTests 'true'
+                            }
+                        }
+                        plugin {
+                            groupId 'org.apache.maven.plugins'
+                            artifactId 'maven-assembly-plugin'
+                            version '2.4'
+                            executions {
+                                execution {
+                                    id 'create-archive'
+                                    phase 'package'
+                                    goals {
+                                        goal 'single'
+                                    }
+                                }
+                            }
+                            configuration {
+                                descriptors {
+                                    descriptor '${project.basedir}/assembly.xml'
+                                }
+                            }
+                        }
+                        plugin {
+                            artifactId 'maven-clean-plugin'
+                            version '3.1.0'
+                            configuration {
+                                filesets {
+                                    fileset {
+                                        directory 'src/main/webapp/docs/'
+                                        includes {
+                                            include 'candlepin-api-spec.yaml'
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }.withXml {
+            dependencyManagement.pomConfigurer.configurePom(asNode())
+            asNode().dependencies.'*'.findAll() {
```

Comment:
This is done because Gradle maven plugin looks for dependency & dependency management tag in ```build.gradle``` file to generate the pom file. In Gradle we have two logical projects (api & candlepin) which we build/compile from single ```build.gradle``` file. In order to auto-generate classes via openapi-generator we have included ```compile project(":api")``` in dependency of main Gradle build. It make sure auto-generated class are available for main build beforehand.

Since :api project is added as a dependency, gradle pom task pick this as a dependency & add that to POM.xml file. We need to remove this dependency from auto-generated pom file.

Reply:
ack

Reply:
Let's explain this in a short comment right above this code

Reply:
Done.